### DISCUSSION TITLE
/dev/tty vs /dev/serial/by-id and stability?

### DISCUSSION THREAD

**Discussion by lurendrejer** [zwave-js/node-red-contrib-zwave-js#167] [Q&A]
Hi.

Just did an observation - I have always been using the path that the plugin suggests /dev/tty* and everything has seemingly been fine.
I removed a handful of devices and wanted to do a full heal of the network, but it failed on multiple nodes every time over a span of several days.
A post on some random zwaveJS post mentioned problems in HA when using /dev/tty instead of /dev/serial/by-id as the path to the controller.
I tried changing my path to (/dev/serial/by-id/usb-0658_0200-if00) - and lo and behold, healing finished in the first run.

Does /dev/tty and /dev/serial handle queues of messages in different ways? what is this?

And, maybe with some further testing - the plugin should show the devices in /dev/serial/ insted of /dev/tty, who knows what other problems might be due to this 'problem' if my experience really just isn't at complete coincidence.

For the record: tried healing upwards of ten times over a span of three days.
tried three times today, the last one with the new path and it suddenly worked.

**Comment by marcus-j-davies** [comment]
Hi @lurendrejer - Hope your well.

I'm not aware of the path being used having an affect on performance, however, I won't be surprised, maybe the OS handles it differently  - @AlCalzone, any thought on this?

For what its worth, I have always used **/dev/serial/by-id/** - as it never changes **/dev/tty*** can change its name after a reboot.

As for listing serial ports, we're not doing this our self, its provided by the **serialports** package - which is cross-platform.
https://github.com/zwave-js/node-red-contrib-zwave-js/blob/e1ac2215eb5bcb76b3c1403ed764038d1a8609df/zwave-js/ui/server.js#L202

**/dev/serial/by-id/** Obviously isn't possible on window, and I want to avoid implementing host specifics if possible, allowing the serialports package to take care of it.

This is the reason I added the custom path option - to be able to do this type of reference if the host provides it.

**Comment by AlCalzone** [comment]
> Does /dev/tty and /dev/serial handle queues of messages in different ways? what is this?

No, they are just symlinks to the same serial device. `serial/by-id` is recommended, because it stays the same even when the stick gets soft-reset (restarted) on driver startup. Older sticks tend to disconnect and reconnect then.

Without seeing driver logs however, I can't say what's going on here.

**Reply by lurendrejer** [reply]
Thank you :)
In that case (symlink) it must have been a complete coincidence, thank you for clearing that out.
